### PR TITLE
Add data encryption

### LINKED ISSUE #60: As a user that shares his/her computer, I would like to encrypt my financial information
so that other users of the computer cannot simply view my financial information by opening the data files

### PR DESCRIPTION
**Pull request by JasonChong96**
resolves #60

### CODE REVIEW

**Comment by JasonChong96** [comment]
Not ready for merging

**Comment by JasonChong96** [comment]
Coverage fall due to impossible to reach lines (IllegalStateException)

File: src/main/java/seedu/address/model/encryption/EncryptedCategory.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package seedu.address.model.encryption;
+
```
Comment (winsonhys) [inline src/main/java/seedu/address/model/encryption/EncryptedCategory.java]: Author name

File: src/main/java/seedu/address/model/encryption/EncryptedCost.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package seedu.address.model.encryption;
+
```
Comment (winsonhys) [inline src/main/java/seedu/address/model/encryption/EncryptedCost.java]: author name

File: src/main/java/seedu/address/model/encryption/EncryptedDate.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package seedu.address.model.encryption;
+
```
Comment (winsonhys) [inline src/main/java/seedu/address/model/encryption/EncryptedDate.java]: author name

File: src/main/java/seedu/address/model/encryption/EncryptedExpense.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
+package seedu.address.model.encryption;
+
```
Comment (winsonhys) [inline src/main/java/seedu/address/model/encryption/EncryptedExpense.java]: author name

File: src/main/java/seedu/address/model/encryption/EncryptedExpenseField.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.encryption;
+
+import static java.util.Objects.requireNonNull;
```
Comment (winsonhys) [inline src/main/java/seedu/address/model/encryption/EncryptedExpenseField.java]: author name

File: src/main/java/seedu/address/model/encryption/EncryptedExpenseTracker.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
+package seedu.address.model.encryption;
+
```
Comment (winsonhys) [inline src/main/java/seedu/address/model/encryption/EncryptedExpenseTracker.java]: authro name

**Pull request merged**